### ISSUE TITLE
Set maximum precision for writing floats

### ISSUE THREAD

**Issue by esahione** [stephenberry/glaze#934]
Hi,

I'm trying to set the write function to be fixed precision for std::float128_t (or long double) values - while I need to store these values with that precision, I only need to output them at 5 decimals. The increased precision slows down the writing relative to rapidjson.

How can I do this in the most efficient fashion?

Thanks!

**Comment by stephenberry** [comment]
For the sake of performance I think it would make sense to cast to a smaller floating point type and then serialize that. I think it makes sense to add a compile time option to Glaze to support this automatically.

Similar to how other Glaze wrappers work, we could make a wrapper that turns on this reduced precision for only specific floating point values or specific arrays or maps.

I think I'll make an enum to indicate the type to cast to:
```c++
enum struct float_precision : uint8_t
{
   none,
   float32,
   float64,
   float128
};
```

Our compile time option will be:
```c++
float_precision float_max_write_precision{}; // the maximum precision type used for writing floats, higher precision floats will be cast down to this precision
```

The associated wrapper if you only want to affect one number/array/object:
```glz::max_write_precision<glz::float_precision::float32, &T::my_number>```

To affect the entire write call you would use:
```c++
constexpr glz::opts options{.float_max_write_precision = glz::float_precision::float32};
glz::write<options>(...);
```

Does this approach seem good to you? Would writing out only 32bit float precision be enough of a reduction. Currently Glaze doesn't support float16_t, but this could be added in the future.

**Comment by esahione** [comment]
Thanks for the fast response! Yes that's great. :)

**Comment by stephenberry** [comment]
Cool, I'll try to add this soon. It shouldn't be that much work.

**Comment by stephenberry** [comment]
Support has been merged (#935).

[Documentation is here](https://github.com/stephenberry/glaze/blob/main/docs/max-float-precision.md)

To make things simpler I added wrappers named `glz::write_float32` and `glz::write_float64`. I don't see the need for a `glz::write_float128` right now, because we don't have larger floating point types. The wrapper `glz::write_float_full` shouldn't normally be used, but it allows you to turn off the precision reduction for specific values if you are reducing the precision at the top level of a write call.
